core/tests/unittests/test_parallel_local_folding.py

```diff
@@ -88,3 +95,10 @@ def test_resource_allocation_and_time_limit():
     for i in range(num_iterations):
         config = searcher.get_config()
         _test_resource_allocation_and_time_limit(**config)
+
+
+@patch("autogluon.common.utils.resource_utils.ResourceManager.get_available_virtual_mem")
+def test_dynamic_resource_allocation(mock_get_mem):
+    mock_get_mem.return_value = 1073741824
+    fold_fitting_strategy = _construct_dummy_fold_strategy(model_base_cls=DummyBigModel, num_jobs=100, num_folds_parallel=8)
+    assert fold_fitting_strategy.num_parallel_jobs == 2
```

Comment:
Can you add a unit test where we are training 8 folds, have memory for 7.5, we train two sets of 4

Add a unit test where we are training 10 folds, we have memory for 6, we train 4, 4, 2. (Also add comment as to why it is doing this)

Reply:
Good point! Added